docs/DeveloperGuide.md

```diff
@@ -283,32 +283,422 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+    * 1b1. Healthcare Xpress shows that patient/nurse.
+
+    * 1b2. Medical administrator requests to edit that patient/nurse and provides the details to edit.
+
+      Use case resumes from step 4.
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. Healthcare Xpress shows an error message.
+
+      Use case resumes at step 2.
+
+* 3b. Medical administrator requests to delete all tags from a patient/nurse.
+
+    * 3b1. Healthcare Xpress deletes all the tags for that patient/nurse.
+  
+      Use case ends. 
+  
+* a. At any time, Medical administrator choose to exit the program.
+
+  Use case ends.
+
+**Use case: UC04 - Add a tag to patient/nurse**
+
+**MSS**
+
+1.  Medical administrator requests to list patients/nurses according to general or specific groups.
+2.  HealthCare Xpress shows a list of patients/nurses that satisfy the group's condition.
+3.  Medical administrator requests to add a tag to a patient/nurse and provides the tag's details.
+4.  HealthCare Xpress add the tag of specific details to the patient/nurse.
+
+    Use case ends.
+
+**Extensions**
+* 1a. The given input is invalid.
+
+    * 1a1. Healthcare Xpress shows an error message.
+
+      Use case restarts.
+
+* 1b. Medical Administrator request to get a specific patient/nurse.
+
+    * 1b1. Healthcare Xpress shows that patient/nurse.
+
+    * 1b2. Medical administrator requests to add tag to the patient/nurse and provides the tag's details.
+
+      Use case resumes from step 4.
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. Healthcare Xpress shows an error message.
+
+      Use case resumes at step 2.
+
+* a. At any time, Medical administrator choose to exit the program.
+
+  Use case ends.
+
+**Use case: UC05 - Delete a specific tag from a patient/nurse**
+
+**MSS**
+
+1.  Medical administrator requests to list patients/nurses according to general or specific groups.
+2.  HealthCare Xpress shows a list of patients/nurses that satisfy the group's condition.
+3.  Medical administrator requests to delete a specific tag from a patient/nurse.
+4.  HealthCare Xpress delete the tag of specific details from the patient/nurse.
+
+    Use case ends.
+
+**Extensions**
+* 1a. The given input is invalid.
+
+    * 1a1. Healthcare Xpress shows an error message.
+
+      Use case restarts.
+
+* 1b. Medical Administrator request to get a specific patient/nurse.
+
+    * 1b1. Healthcare Xpress shows that patient/nurse.
+
+    * 1b2. Medical administrator requests to delete tag of specific details from that patient/nurse.
+
+      Use case resumes from step 4.
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. Healthcare Xpress shows an error message.
+
+      Use case resumes at step 2.
+
+* a. At any time, Medical administrator choose to exit the program.
+
+  Use case ends.
+
+**Use case: UC06 - Add a date and time to patient**
```

Comment:
Just a small suggestion but maybe we can name this "home visit date and time" instead of just "date and time"

Reply:
Okay will change